keps/sig-network/5343-kube-proxy-backends/README.md

```diff
@@ -0,0 +1,824 @@
+might do things like assume the existence of certain `iptables`
+chains). When we [changed the default proxy mode from `userspace` to
+`iptables`] (in Kubernetes 1.2), we decided to just not worry about
+compatibility problems. We can't plausibly do that this time.
+
+Rather than actually changing the default, we could try to get to a
+point where the "default" no longer matters, because everyone is
+specifying which proxy mode they want *explicitly*. We can start by
+having kube-proxy log warnings and Events if you don't explicitly
+specify a proxy mode, and eventually, after enough time has passed, we
+could make it an error to start kube-proxy without explicitly
+specifying the mode.
+
+Assuming the `v1alpha2` config file format happens, we could also
+remove defaulting of `mode` from that config version, and make the
+`v1alpha2` `mode` field be required and non-defaulted.
+
+Another possibility would be to deprecate the existing multi-mode
+kube-proxy binary in favor of having separate `kube-proxy-iptables`,
+`kube-proxy-ipvs`, and `kube-proxy-nftables` binaries (and perhaps,
+eventually, separate images). That would also work well with the plan
+to deprecate `ipvs` mode (and would allow us to completely remove
+the existing deprecated CLI options)...
+
+[changed the default proxy mode from `userspace` to `iptables`]: https://github.com/kubernetes/kubernetes/pull/16344
+
+### Deprecating the `ipvs` backend
+
+We will have to maintain both `nftables` and `iptables` for a while,
+but we would like to stop maintaining `ipvs` as soon as we can.
+
+At this point, there is no real reason to prefer the `ipvs` backend to
+`nftables`, other than it having slightly looser kernel version
+requirements. The `nftables` backend has slightly better performance
+than `ipvs` (plus a cleaner architecture and better feature parity
+with `iptables`) and none of the other features that make IPVS
+interesting (such as configurable schedulers) really apply to the case
+of a Kubernetes service proxy. (The simpler schedulers, like `rr`,
+could be implemented in the `nftables` backend if we cared, while the
+more complicated ones, like `mh`, end up being thwarted by the fact
+that the kube-proxy instances on different nodes don't share
+information with each other.)
+
+At this point, we are not likely to add support for new features to
+the `ipvs` backend (unless they come "for free" via the
+backend-indepdent code). We have also slowed down bugfixing for
+`ipvs`, and suggested that users filing bugs against it should try out
+the `nftables` backend instead.
+
+At some point we will need to take some more definitive action.
+Possibilities include:
+
+  - Logging warnings/Events about the fact that `ipvs` mode is
+    deprecated.
+
+  - Moving the `ipvs` mode out of the primary `kube-proxy` binary into
+    a separate `kube-proxy-ipvs` binary.
+
+  - Moving `kube-proxy-ipvs` to a staged repository.
+
+  - Moving `kube-proxy-ipvs` to a non-staged repository.
```

Comment:
I will just fork the entire code in its own repo and open it for new maintainers